Formula/atop.rb

```diff
@@ -0,0 +1,27 @@
+class Atop < Formula
+  desc "Advanced system and process monitor for Linux using process events"
+  homepage "https://www.atoptool.nl"
+  url "https://github.com/Atoptool/atop/archive/refs/tags/v2.6.0.tar.gz"
+  sha256 "134de16cd58fb651d952e8ac401b20fe7f724afc0b152f06486e5c6f7be9c31a"
+  license "GPL-2.0-or-later"
+  head "https://github.com/Atoptool/atop.git", branch: "master"
+
+  depends_on :linux
+  depends_on "ncurses"
+  depends_on "zlib"
+
+  def install
+    if build.head?
+      inreplace "version.h" do |s|
+        s.sub!(/"$/, "-#{Utils.git_head}\"")
+      end
+    end
+    system "make"
+    bin.install "atop"
+    man1.install gzip("man/atop.1")
+  end
+
+  test do
+    assert_match "Version:", shell_output("#{bin}/atop -V")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks, added several test cases testing the functionality now.